src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/FluxCommunication2.hpp

```diff
@@ -0,0 +1,352 @@
+      const auto& direction = direction_neighbors.first;
+      const size_t dimension = direction.dimension();
+      const auto& neighbors_in_direction = direction_neighbors.second;
+      ASSERT(neighbors_in_direction.size() == 1,
+             "h-adaptivity is not supported yet.\nDir: "
+                 << direction << "\nDim: " << dimension << "\nNeighbors:\n"
+                 << neighbors_in_direction);
+      const auto& orientation = neighbors_in_direction.orientation();
+      const auto boundary_extents = extents.slice_away(dimension);
+      auto boundary_variables = db::data_on_slice(
+          box, extents, dimension,
+          direction.side() == Side::Lower ? 0 : extents[dimension] - 1,
+          tmpl::remove_duplicates<
+              tmpl::append<typename Metavariables::normal_dot_numerical_flux::
+                               type::slice_tags,
+                           typename Metavariables::system::normal_dot_fluxes::
+                               argument_tags>>{});
+
+      // Everything in the loop over neighbors needs to be fixed for
+      // hp-adaptivity to handle projections correctly
+      for (const auto& neighbor : neighbors_in_direction) {
+        const auto& face_normal =
+            db::get<Tags::UnnormalizedFaceNormal<volume_dim>>(box).at(
+                direction);
+        DataVector magnitude_of_face_normal = magnitude(face_normal);
+        std::decay_t<decltype(face_normal)> unit_face_normal(
+            magnitude_of_face_normal.size(), 0.0);
+        for (size_t d = 0; d < volume_dim; ++d) {
+          unit_face_normal.get(d) =
+              face_normal.get(d) / magnitude_of_face_normal;
+        }
+
+        // This needs to be updated for conservative systems where we can just
+        // slice the flux from the volume.
+        NormalDotFluxType local_boundary_flux(face_normal.begin()->size(), 0.0);
+        apply_flux<Metavariables>(
+            make_not_null(&local_boundary_flux), boundary_variables,
+            unit_face_normal,
+            typename system::normal_dot_fluxes::argument_tags{});
+
+        // We compute the parts of the numerical flux that only depend on data
+        // from this side of the mortar now, then package it into a Variables.
+        // We store one copy of the Variables and send another, since we need
+        // the data on both sides of the mortar.
+        PackagedData packaged_data(face_normal.begin()->size(), 0.0);
+        get_packaged_data_from_numerical_flux(
+            make_not_null(&packaged_data), normal_dot_numerical_flux_computer,
+            local_boundary_flux, boundary_variables,
+            typename Metavariables::normal_dot_numerical_flux::type::
+                slice_tags{},
+            unit_face_normal);
+
+        const auto direction_from_neighbor = orientation(direction.opposite());
+
+        // This only needs to be created in the case where the data is oriented
+        // differently.
```

Comment:
add this will be improved later (also note that adaptivity does not affect this)